routes/warehousesRoutes.js

```diff
@@ -0,0 +1,31 @@
+const express = require('express');
+const router = express.Router();
+const knex = require('../knexfile');
+
+// DELETE /api/warehouses/:id
+router.delete('/:id', async (req, res) => {
+  const warehouseId = req.params.id;
+
+  try {
+    const warehouse = await knex('warehouses').where('id', warehouseId).first();
+
+    if (!warehouse) {
+      return res.status(404).send({
+        message: "Warehouse not found",
+      });
+    }
+
+    // Delete the warehouse with the associated inventories
+    await knex('warehouses')
+      .where('id', warehouseId)
+      .del();
+
+    return res.status(204).send();
```

Comment:
While using res.status(204).send() is technically correct for indicating a successful operation with no content, it’s important to consider whether this provides enough context for the client. You should always aim to send a clear and valid response that gives the client more information about the outcome of the operation. For example, a 200 status with a message could be more informative.


`return res.status(200).json({ message: "Warehouse Deleted successfully." });
`

Reply:
that's a great point thank you! I've updated the messages now